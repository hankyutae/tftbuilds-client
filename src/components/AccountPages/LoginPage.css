.login-box {
  
  z-index:2;
  height: 500px;
  width: 300px;/* 
  background: rgba(200, 200, 200, 0.88); */

  position: absolute;
  top:150px;
  
  /* margin-top: -250px; */
}


.offset1{
  stop-color:rgb(199, 175, 89);
  stop-color:#FFF0A2;
  stop-opacity:1;
}

.offset2{
  stop-color:rgb(255,215,0);
  stop-opacity:1;
}


/* .svg-gold{
} */
.svg-div{
  padding-top:70px;
  padding-left:10px;
  padding-bottom:20px;
}
.hex-gold{
  fill:rgba(255,215,0,0.6);
  fill:#FBAF47;
}
.hex-black{
  
  fill:rgba(211, 211, 211, 0.6);/* 
  fill:rgb(34, 33, 33,0.7); */
}

.login-box input{
  border-radius:5px;
  border:0;
}

@media screen and (min-width:300px){
  .login-box{
    left: calc(50% - 150px);
  }
}

@media screen and (max-width:300px){
  main{
    position:relative;
    overflow-x:scroll;
  }
}

.login-box{
padding:12px 10px;
}

.login-box .fieldset-style{
padding:0;
margin:0;
text-align:left;
border:0;
} 

.login-box .fieldset-style > legend{
margin-bottom: 2%;
}

.login-box input{
width:100%;
}
.login-box .login-sign-in{
  width:100%;
  border:0px;
  padding:0.5rem;
  border-radius:5px;
  background-color:#A2B9B1;
  background-color:#e8e5f3;
  font-weight:550;
}


@media (hover:hover){
  .login-box .login-sign-in:hover{
    background-color:rgba(162,185,177,0.8);
  }
  .create-account-from-login:hover{
    background-color:rgba(0,0,0,0.4);
  
  }
}
.login-box input{
  padding:0.5rem;
}

.login-box .login-helper{
text-align:left;
}
.login-box .forgot-login{
float:left;
}
.login-box .login-signup-link{
float:right;
}
.login-box h1.login-box-logo{
padding:0;
margin:0;/* 
position:absolute;
top: 20px;
left:67px; */
margin-bottom:25px;
}
.red{color:red;}

/* @media screen and (max-height:538px){
  main{
    min-height:700px;
  }
} */