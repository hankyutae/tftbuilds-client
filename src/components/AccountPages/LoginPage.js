import React from 'react';
import {Link} from 'react-router-dom';
import './LoginPage.css';
import TokenService from '../../services/token-service';
import AuthApiService from '../../services/auth-api-service';
//import TftContext from '../../contexts/TftContext';

export default class LoginPage extends React.Component{

  state={error:null}
  //static contextType = TftContext
  handleSubmitJwtAuth = ev => {
    ev.preventDefault()
    this.setState({ error: null })
    const { user_name, password } = ev.target

    AuthApiService.postLogin({
      user_name: user_name.value,
      password: password.value,
    })
      .then(res => {
        user_name.value = ''
        password.value = ''
        TokenService.saveAuthToken(res.authToken)
        this.onLoginSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }


  onLoginSuccess = () => {
    const { /* location, */ history } = this.props;/* 
    const destination = (location.state || {}).from || '/'; */
    this.props.propForComponent();/* 
    history.push(destination) */
    history.goBack();
  }


  render(){
    const { error } = this.state;
    return(<>
      <div className='svg-div'>
        
        <svg className='svg-gold' width="310" height="600">
          
          {/* <polyline className='hex-gold' points="150,0 300,87 300,260 150,346 0,260 0,87 150,0"/> */}
          
          <defs>
    <radialGradient id="grad1" cx="50%" cy="50%" r="90%" fx="50%" fy="50%">
      <stop offset="0%" className='offset1'/>
      <stop offset="100%" className='offset2'/>
    </radialGradient>
  </defs>
          <polyline className='hex-black' points="150,0 300,87 300,97 150,10 0,97 0,87 150,0"/> />
          <polyline className='hex-black' points="150,556 300,470 300,480 150,566 0,480 0,470 150,556"/>
          <polygon /* className='hex-gold' */ points="150,10 300,97 300,470 150,556 0,470 0,97 150,10" fill="url(#grad1)"/>
        </svg>
      </div>
      
      
      <form className='login-box'  onSubmit={this.handleSubmitJwtAuth}></form>

      <form className='login-box'  onSubmit={this.handleSubmitJwtAuth}>
      
      
      <h1 className="login-box-logo">
        Login
      </h1>
      <fieldset className='fieldset-style'>
          {/* <legend>Sign in </legend> */}
          <div role='alert'>
          {error && <p className='red'>{error}</p>}
          </div>
          <p>
            <label htmlFor="userid">Username</label><br/>
            <input type="text" name="user_name" id="userid"/>
          </p>
          <p>
            <label htmlFor="password">Password</label><br/>
            <input type="password" name="password" id="password"/>
          </p>
      </fieldset>
      <button className="login-sign-in" type='submit'> Sign in </button> <br/>
      <div className="login-helper">
        {/* <div className="forgot-login">
          <p>
            <a>Forgot username?</a>
          </p>
          <p>
            <a>Forgot password?</a>
          </p>
        </div> */}
        <div className="login-signup-link">
          <p>
            <Link className='create-account-from-login' to='/signup'>Create account?</Link>
          </p>
        </div>
      </div>
    </form>
</>
    );
  }
}
