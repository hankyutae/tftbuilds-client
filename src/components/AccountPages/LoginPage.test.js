import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import LoginPage from './LoginPage';
import renderer from "react-test-renderer";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <LoginPage/>
    </BrowserRouter>
  , div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders the UI as expected", () => {
  const tree = renderer
    .create( 
      <BrowserRouter>
        <LoginPage/>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});