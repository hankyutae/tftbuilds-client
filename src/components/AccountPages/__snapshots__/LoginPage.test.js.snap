// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <div
    className="svg-div"
  >
    <svg
      className="svg-gold"
      height="600"
      width="310"
    >
      <defs>
        <radialGradient
          cx="50%"
          cy="50%"
          fx="50%"
          fy="50%"
          id="grad1"
          r="90%"
        >
          <stop
            className="offset1"
            offset="0%"
          />
          <stop
            className="offset2"
            offset="100%"
          />
        </radialGradient>
      </defs>
      <polyline
        className="hex-black"
        points="150,0 300,87 300,97 150,10 0,97 0,87 150,0"
      />
       /&gt;
      <polyline
        className="hex-black"
        points="150,556 300,470 300,480 150,566 0,480 0,470 150,556"
      />
      <polygon
        fill="url(#grad1)"
        points="150,10 300,97 300,470 150,556 0,470 0,97 150,10"
      />
    </svg>
  </div>,
  <form
    className="login-box"
    onSubmit={[Function]}
  />,
  <form
    className="login-box"
    onSubmit={[Function]}
  >
    <h1
      className="login-box-logo"
    >
      Login
    </h1>
    <fieldset
      className="fieldset-style"
    >
      <div
        role="alert"
      />
      <p>
        <label
          htmlFor="userid"
        >
          Username
        </label>
        <br />
        <input
          id="userid"
          name="user_name"
          type="text"
        />
      </p>
      <p>
        <label
          htmlFor="password"
        >
          Password
        </label>
        <br />
        <input
          id="password"
          name="password"
          type="password"
        />
      </p>
    </fieldset>
    <button
      className="login-sign-in"
      type="submit"
    >
       Sign in 
    </button>
     
    <br />
    <div
      className="login-helper"
    >
      <div
        className="login-signup-link"
      >
        <p>
          <a
            className="create-account-from-login"
            href="/signup"
            onClick={[Function]}
          >
            Create account?
          </a>
        </p>
      </div>
    </div>
  </form>,
]
`;
